spec/actors/hyku_addons/actors/note_field_actor_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe HykuAddons::Actors::NoteFieldActor do
+  let(:rand) { SecureRandom.uuid }
+  let(:attributes) { {} }
+  let(:user) { build(:user) }
+  let(:ability) { ::Ability.new(user) }
+  let(:work) { create(:generic_work) }
+  let(:env_class) { Hyrax::Actors::Environment }
+  let(:env) { env_class.new(work, ability, attributes) }
+  let(:terminator) { Hyrax::Actors::Terminator.new }
+
+  # let(:actor) { described_class.new(terminator) }
+
+  # This lets us use the middleware chain, rather than assuming its running and using the `create` method on the actor
+  let(:middleware) do
+    stack = ActionDispatch::MiddlewareStack.new.tap do |middleware|
+      middleware.use described_class
+    end
+
+    stack.build(terminator)
+  end
+
+  describe "#create" do
+    context "when the note attribute is present" do
+      let(:attributes) { { note: 'this is a note' } }
+
+      before do
+        allow(terminator).to receive(:create).with(env)
+        allow(middleware).to receive(:serialize_note_field).with(env_class)
+      end
+
+      it "call serialize_note_field method while creating work" do
+        middleware.create(env)
+        expect(middleware).to have_received(:serialize_note_field).with(env)
+      end
+
+      it "returns an ActiveTriples::Relation class for note attribute" do
+        middleware.create(env)
+        expect(work.note.class).to eq(ActiveTriples::Relation)
+      end
```

Comment:
I think it would be more helpful to see that the note got serialized as JSON and persisted on the work.  This might look like:
```
it "adds the note" do
  expect(work.note).to be_blank
  middleware.create(env)
  expect(work.note).to be_present
  expect(JSON.parse(work.note.first)).to include(email: user.email, timestamp: String, note: note)
end
```